jgt_common/futures.py

```diff
@@ -0,0 +1,190 @@
+"""
+Helpful functions for using concurrent.futures with a shared thread pool executor.
+
+Terminology:
+
+  * fdict - futures dictionary - as in the examples for the concurrent.futures
+    module documentation, a dictionary whose keys are futures and whose values
+    are related to the work being done. See each function for details on what
+    the values will be.
+
+Purpose:
+
+   These helpers are meant to be used for "spot" parallelism,
+   not for whole application parallelism. In places where the code needs to do
+   a whole bunch of slow things that can be done in parallel (API calls, etc)
+   and then "goes synchronous again."  (Testing is one such use case.)
+
+
+Archicture:
+
+   One executor thread pool is used by all of these functions.
+   To support the use of these functions in a "library" kind of way,
+   configuring the size of the thread pool and shutting it down (if ever used)
+   are done by calling separate functions. This is so an application can use
+   configuration data for the set up without having to spread knowledge of that
+   configuration data to any function(s) that want to use these functions.
+
+   The executor thread pool defined here is mainly meant for use by the other
+   functions defined here, but users of this module are free to use it themselves
+   directly as needed.
+
+"""
+
+from concurrent.futures import ThreadPoolExecutor as _ThreadPoolExecutor
+from concurrent.futures import as_completed  # noqa - imported for pass-through use.
```

Comment:
Is the `#noqa` still needed here since it's called multiple times below?

Reply:
I suppose not?  It's not imported with an underscore prefix for pass-through-ness, but that could be a plain old (no noqa) comment. WF.

Reply:
Ah, I see what you mean. Yeah, I think the comment is fine but no need for the `noqa`